src/instructlab/eval/ruler.py

```diff
@@ -0,0 +1,161 @@
+# Standard
+from typing import Any, Dict, List, Optional
+import json
+import os
+import pathlib
+
+# Third Party
+from lm_eval.evaluator import simple_evaluate
+
+# First Party
+from instructlab.eval.evaluator import Evaluator
+
+RULER_TASKS = [
+    "niah_single_1",
+    "niah_single_2",
+    "niah_single_3",
+    "niah_multikey_1",
+    "niah_multikey_2",
+    "niah_multikey_3",
+    "niah_multiquery",
+    "niah_multivalue",
+    "ruler_vt",
+    "ruler_cwe",
+    "ruler_fwe",
+    "ruler_qa_hotpot",
+    "ruler_qa_squad",
+]
+
+
+class RulerEvaluator(Evaluator):
+    """
+    Class definition for running RULER benchmarking tasks.
+    """
+
+    def __init__(
+        self,
+        model_path: Optional[str] = None,
+        output_file: Optional[str] = None,
+        tasks: list[str] = RULER_TASKS,
+        num_gpus: Optional[int] = None,
+        eval_config: Optional[Dict[str, Any]] = None,
+        vllm_config: Optional[Dict[str, Any]] = None,
+        hf_config: Optional[Dict[str, Any]] = None,
+        openai_config: Optional[Dict[str, Any]] = None,
```

Comment:
I see that you've declared this as an argument and are reading it further down, but I don't see it actually being passed into lm-eval-harness. You'll want to make sure we're doing this by following how I did it here: https://github.com/instructlab/eval/blob/f48c6a1833d98e0bb3cf8931e587ce5102c686a6/src/instructlab/eval/leaderboard.py#L745

Reply:
got rid of these for now, since they werent being used